vlm/src/main/scala/geotrellis/contrib/vlm/RasterViewOptions.scala

```diff
@@ -0,0 +1,13 @@
+package geotrellis.contrib.vlm
+
+import geotrellis.proj4.CRS
+import geotrellis.vector.Extent
+import geotrellis.raster.{RasterExtent, CellType, Raster, MultibandTile}
+
+
+private[vlm] case class RasterViewOptions(
+  crs: Option[CRS] = None,
+  cellType: Option[CellType] = None,
+  rasterExtent: Option[RasterExtent] = None,
+  readMethod:  Option[(Extent, Seq[Int]) => Option[Raster[MultibandTile]]] = None
```

Comment:
I'm concerned about the effect on serialization attaching a this type of function to each `RasterSource` has. Its potentially easy to close over some non-serializable resource without realizing. In current implementation this would close over a reference to the source `RasterSource` and link to to converted `RasterSource`. Which isn't critical failure but is not great. Is there a another easier way to achieve the same thing here?